### ISSUE TITLE
BBB 2.5: SIP-Dialin not waiting for moderation

### ISSUE THREAD

**Issue by jacotec** [bigbluebutton/bigbluebutton#15163]
**Describe the bug**
After going from 2.4.8 to 2.5.0 I found an issue with telephone-dialin participants:

The room is set for moderation. When someone calls the dialin-number and enters the PIN, the moderator sees the new "waiting participant".

Although the moderator did not accept the waiting participant, the caller can fully participate to the conference - he hears the others in the conference and he can talk.

In addition, the dialin participants now are always displayed with the dialin-number instead of their real caller-ID

**To Reproduce**
1. Open a room on a BBB server with a configured SIP-Dialin
2. The room is set for moderation (the moderator must approve guests)
3. Dial-in via phone
4. A new "Waiting participant" is signalled to the moderator
5. At this point the caller should not be able to listen to the conference and talk to the conference! But he is already able to do both, although he's still in the moderation queue.
6. The caller is shown with the dial-in number instead of its caller-id. If the dial-in number of the server is 0221123456 and the caller-id of the calling phone is 01729876543, in 2.4.8 the participant is correctly shown as "01729876543". In 2.5.0 instead the participant is shown as "0221123456" which is the dial-in number of BBB. So, with 5 phone participants, all 5 are shown with the same, wrong number.

**Expected behavior**
As long as a call-in participant is in the waiting queue for moderation, he must not be able to listen and talk to the conference.
A call-in participant must be shown with his caller-id in the moderation queue and user list.

**Actual behavior**
Straight after entering the PIN, the caller can listen and talk to the conference although he is not yet approved by the moderator
The participant is always shown with the dial-in phone number of the server instead which his real caller-id.

**Screenshots**
![image](https://user-images.githubusercontent.com/3321655/173056128-f3b02a5b-8a77-4832-8323-01f0ba77b45f.png)

**Desktop (please complete the following information):**
 - OS: [e.g. Windows, Mac] Windows, Mac
 - Browser [e.g. Chrome, Safari] Edge, Safari

**Smartphone (please complete the following information):**
N/A

**Comment by zhem0004** [comment]
I tried to reproduce it, but it works correctly locally

![Screenshot from 2022-06-10 11-49-00](https://user-images.githubusercontent.com/80958763/173103800-4f463d3a-360b-4f06-afa0-c65018df3fcc.png)

**Comment by schrd** [comment]
I was able to reproduce parts of it on 2.5.1:

1. setup the room for ask moderator guest policy
2. dial in
3. observations:
    * participant is immediately bridged to the conference
    * participant is shown in the waiting users list
    * participant is not shown in the users list
    * participants name is the one that is set by the sip trunk provider

In 2.4.8 the behaviour was different:

1. setup the room for as moderator guest policy
2. dial in
3. observations:
    * participant is immediately bridged to the conference
    * participant does not appear in the waiting users list
    * participant is show in the users list as dial in user
    * participants name is the one that is set by the sip trunk provider

I think the name/phone number issue is something different. @jacotec did you try this on the same server? I am asking because each server usually has its own SIP trunk set up. So the SIP trunk might be set up differently.

You can lookup the user name/phone number of the participant that is sent by the telephone provider in freeswitch logs. BBB should use that one.

**Comment by maximilianhoecker** [comment]
Using 2.5.0 can reproduce exactly, what @schrd and @jacotec described:

Steps:

1. Open a room having moderator guest policy activated
2. Dialin from a mobile phone using number + pin

**Expected behavior (after dialing + entering pin):**
- Dialin-user hears nothing or waiting music (prefered)
- Approved Room-Participants should not hear the dialin-users in the waiting room
- Dialin-user should never hear something from the room participants
- Moderator gets notified that the dialin user is waiting in the waiting room and needs to get approved
- Voicebridge should **not** put the user into the same context as the rest of the conference participants until the moderator approved the dialin user

**Actual behavior (after dialing + entering pin):**
- Dialin-user hears all other room participants
- Approved Room-Participants hear the dialin-users in the waiting room
- Moderator gets notified that the dialin user is waiting in the waiting room and needs to get approved
- Voicebridge puts the user into the same context as the rest of the conference participants

In summary this is a security leak, concering the originally intended function of the waiting room for dialin users.

For 2.4.8 for me this works like designed.

**Comment by maximilianhoecker** [comment]
@schrd As far as I understood the PR #14344 a behavior of at least "not letting a dialin user hear something from the conference" was intended (but maybe I am wrong with the interpretation of the phrase _"Phone user's ability to join the meeting depends on guest policy set by moderator"_ in the PR description?)

(independend of my capabilities in understanding the PR:) Thanks @zhem0004 for the mentioned PR by the way 👏 👍

**Comment by schrd** [comment]
Ok the missing part seems to be in controlling freewitch:

* setting a participant to deaf: `conference <number> deaf <participant number>`
* mute participant: `conference <number> mute <participant number>`
* playing a sound to the participant that he is waiting for moderator approval: `conference <number> play <filename> <participant number>`
* unmute participant: `conference <number> unmute <participant number>`
* undeaf participant: `conference <number> undeaf <participant number>`

**Comment by jacotec** [comment]
> I think the name/phone number issue is something different. @jacotec did you try this on the same server? I am asking because each server usually has its own SIP trunk set up. So the SIP trunk might be set up differently.
> 
> You can lookup the user name/phone number of the participant that is sent by the telephone provider in freeswitch logs. BBB should use that one.

I agree to this. I've updated a second server and here the caller ID works. Strang: It's the same provider as on my first server and all files below /opt/freeswitch are identical except the different SIP login data and the server's IP address.

I'll see if the freeswitch logs will help me. In 2.4.8 the caller ID was fine on both servers using the same config files.

**Comment by mariogasparoni** [comment]
I can confirm this happens for 2.5 (except for the problem with the caller name). Marking as accepted.

**Comment by hostbbb** [comment]
I'll take a look at prototyping this..  the dialplan needs to look into mongo to see if meeting is a moderator approval required, if it it throw the user on hold playing message, awaiting approval,   then when the user is approved, move the user to the conference.
